sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -47,12 +47,20 @@ case class UserDefinedFunction protected[sql] (
     dataType: DataType,
     inputTypes: Option[Seq[DataType]]) {
 
+  // Optionally used for printing UDF names in EXPLAIN
+  private var nameOption: Option[String] = None
```

Comment:
@rxin How about this fix? https://github.com/apache/spark/pull/17712/files#diff-989058ad6f2f8c863fff5e2e07eddaa0R72